### PR TITLE
Kubernetes and rook-operator README

### PR DESCRIPTION
**Pull request by jbw976**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,146 +38,108 @@ Please help us by [Contributing](CONTRIBUTING.md) to the project.
...
+osd-rth3q                       1/1       Running   0          1m
+rgw-1785797224-9xb4r            1/1       Running   0          1m
+rgw-1785797224-vbg8d            1/1       Running   0          1m
+rook-api-4184191414-l0wmw       1/1       Running   0          1m
+rook-operator-349747813-c3dmm   1/1       Running   0          1m
+```
 
-3. Get the connection information for accessing object storage
+#### Provision Storage
+Before Rook can start provisioning storage, a StorageClass needs to be created. This is used to specify the storage privisioner, parameters, admin secret and other information needed for Kubernetes to interoperate with Rook for provisioning persistent volumes.
```
Comment (DanKerns) [inline README.md]: privisioner => provisioner

File: README.md [hunk 2]
```diff
@@ -34,10 +39,109 @@ Please help us by [Contributing](CONTRIBUTING.md) to the project.
...
 
-### Linux
+### Kubernetes
+
+This example shows how to build a simple, multi-tier web application on Kubernetes using persistent volumes enabled by Rook.
+
+#### Prerequisites
+
+This example requires a running Kubernetes cluster. To make sure you have a Kubernetes cluster that is ready for `rook`, you can [follow these quick instructions](demo/kubernetes/README.md).
+
```
Comment (bassam) [inline README.md]: Would be good to highlight that we are looking for a way to make this work out of the box on K8S (and possibly creating a tracking issue in the kubernetes repo).

File: README.md [hunk 3]
```diff
@@ -34,10 +39,109 @@ Please help us by [Contributing](CONTRIBUTING.md) to the project.
...
+mon1                            1/1       Running   0          1m
+mon2                            1/1       Running   0          1m
+osd-n1sm3                       1/1       Running   0          1m
+osd-pb0sh                       1/1       Running   0          1m
+osd-rth3q                       1/1       Running   0          1m
+rgw-1785797224-9xb4r            1/1       Running   0          1m
+rgw-1785797224-vbg8d            1/1       Running   0          1m
+rook-api-4184191414-l0wmw       1/1       Running   0          1m
+rook-operator-349747813-c3dmm   1/1       Running   0          1m
+```
```
Comment (bassam) [inline README.md]: rgw should not be created at this level, no? If its a temporary I would make a note of it and reference the issue open around it.

File: README.md [hunk 4]
```diff
@@ -34,10 +39,109 @@ Please help us by [Contributing](CONTRIBUTING.md) to the project.
...
+
+#### Provision Storage
+Before Rook can start provisioning storage, a StorageClass needs to be created. This is used to specify information needed for Kubernetes to interoperate with Rook for provisioning persistent volumes.  Rook already creates a default admin and demo user, whose secrets are already specified in the sample [rook-storageclass.yaml](demo/kubernetes/rook-storageclass.yaml).
+
+Now we just need to specify the Ceph monitor endpoints (requires `jq`):
+
+```
+export MONS=$(kubectl -n rook get pod mon0 mon1 mon2 -o json|jq ".items[].status.podIP"|tr -d "\""|sed -e 's/$/:6790/'|paste -s -d, -)
+sed 's#INSERT_HERE#'$MONS'#' rook-storageclass.yaml | kubectl create -f -
```
Comment (bassam) [inline README.md]: also make a point there that we are going to streamline this in 0.4 and point at the issue.

File: demo/kubernetes/README.md [hunk 5]
```diff
@@ -1,170 +1,25 @@
-# Rook on Kubernetes
 
-A demo of persistent block storage from rook within a [Kubernetes](http://kubernetes.io/) cluster. We will use a mysql service in Kubernetes as an example of using block storage from rook.
+## Existing Kubernetes Cluster
+If you already have a running Kubernetes cluster, you will also need to modify the kubelet service to bind mount `/sbin/modprobe` to allow access to `modprobe`. Access to modprobe is necessary for using the rbd volume plugin (<https://github.com/kubernetes/kubernetes/issues/23924>).
+If using RKT, you can allow modprobe by following this [doc](https://github.com/coreos/coreos-kubernetes/blob/master/Documentation/kubelet-wrapper.md#allow-pods-to-use-rbd-volumes).
```
Comment (bassam) [inline demo/kubernetes/README.md]: can we please repeat the instructions needed to on this page?

**Comment by bassam** [comment]
Going forward, I think we should simplify the README page maybe into some smaller quickstart sections.  This is great for 0.3.

File: README.md (hunk 3 above)
Comment (jbw976) [inline README.md]: Added a note and opened issue #413 for that (i didn't see an existing issue)

File: README.md (hunk 2 above)
Comment (jbw976) [inline README.md]: How does this note sound?
> Note that we are striving for even more smooth integration with Kubernetes in the future such that `rook` will work out of the box with any Kubernetes cluster.

Comment (jbw976) [inline README.md]: @kokhang i would appreciate if you could open any tracking issues that we know of for Kubernetes cluster bootstrapping and being `rook` ready.

File: README.md (hunk 4 above)
Comment (jbw976) [inline README.md]: Added a note that mentions #355

File: demo/kubernetes/README.md (hunk 5 above)
Comment (jbw976) [inline demo/kubernetes/README.md]: copied the instructions inline for convenience

**Pull request merged**